react-app-drawer-component/my-app/src/AppDrawer.jsx

```diff
@@ -0,0 +1,45 @@
+import { useState } from 'react';
+import { GiHamburgerMenu } from 'react-icons/gi';
+
+export default function AppDrawer({ list, onSelect }) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  function toggleOpen() {
+    setIsOpen(!isOpen);
+  }
+
+  function displayItem(item) {
+    toggleOpen();
+    onSelect(item);
+  }
+  return (
+    <>
+      <GiHamburgerMenu onClick={toggleOpen} className="menu-icon" />
+      <Gradient isDrawn={isOpen} onClick={toggleOpen} />
+      <Menu isOpen={isOpen}>
+        <h3 className="menu-heading">'Choose a Game'</h3>
+        <ul className="menu-items">
+          {list.map((item) => (
+            <li
+              key={item}
+              className="menu-item"
+              onClick={() => displayItem(item)}
+              style={{ color: 'black' }}>
+              {item}
+            </li>
+          ))}
+        </ul>
+      </Menu>
+    </>
+  );
+}
+
+function Menu({ isOpen, children }) {
```

Comment:
Where did you get this solution? I did not demo it this way in the Q&A, and we haven't learned about `children`. But it is like this in the instructor's solution.

I like your solution, I just want to make sure you are understanding the solutions you create. If you delete all this and do it again on your own, can you recreate it?

Reply:
This came from when you told us to take a screenshot if we needed to look back to. I will practice more by redoing the exercise.

Reply:
Oh, thanks! I remembered solving it in front of everyone but I didn't remember that I showed everyone that full solution. When I saw this same approach by others, I realized my mistake and came back here to apologize!